crates/ere-sp1/src/compile.rs

```diff
@@ -1,102 +1,53 @@
-use std::{fs, path::Path, process::Command};
+use std::{path::Path, process::Command};
 
+use build_utils::docker;
 use tempfile::TempDir;
-use toml::Value as TomlValue;
 use tracing::info;
 
 use crate::error::CompileError;
 
-/// Compile the guest crate and return raw ELF bytes.
-pub fn compile_sp1_program(program_crate_path: &Path) -> Result<Vec<u8>, CompileError> {
```

Comment:
All this got simplified.

Reply:
This was mainly here because different zkVMs have different ways to compile the program -- for SP1 it was straightforward because they have a CLI that could be used that also allows us to specify an output directory, when I was integrating the other zkVMs, some did not allow this

Reply:
@jsign we can see what it looks like with the others implemented, I think we may need to bring this back and put a CLI over it and then this is what gets dockerized

Reply:
@kevaundray, ah, I see what you mean. Other zkVMs having more complex guest-compilation flows. Good point.

Uhm, maybe instead of dockerizing these Rust-like programs, we can simply make a `compile_guest_sp1.sh` scripts running exactly the same thing (without all the Rust overhead), and we mount also those scripts and simply execute the script.

So in the `compile_guest_xxx.sh` script we can be free to do whatever magic is needed for compiling guest programs as if you were doing it in a shell.